ui/selectors/selectors.js

```diff
@@ -690,9 +690,11 @@ export function getKnownMethodData(state, data) {
   }
   const prefixedData = addHexPrefix(data);
   const fourBytePrefix = prefixedData.slice(0, 10);
-  const { knownMethodData } = state.metamask;
-
-  return knownMethodData && knownMethodData[fourBytePrefix];
+  const { knownMethodData, use4ByteResolution } = state.metamask;
+  // If 4byte setting is off, we do not want to return the knownMethodData
+  return (
+    use4ByteResolution && knownMethodData && knownMethodData[fourBytePrefix]
```

Comment:
Improvement idea:

```
use4ByteResolution && knownMethodData?.[fourBytePrefix]
 
```

Reply:
didn't know you could nullcheck into an array index, thanks!